tests/integration/test_sqlalchemy_integration.py

```diff
@@ -0,0 +1,259 @@
+                                connect_args={"source": "test", "max_attempts": 1})
+    yield engine, engine.connect()
+
+
+@pytest.mark.parametrize('trino_connection', ['tpch'], indirect=True)
+def test_select_query(trino_connection):
+    _, conn = trino_connection
+    metadata = sqla.MetaData()
+    nations = sqla.Table('nation', metadata, schema='tiny', autoload_with=conn)
+    assert_column(nations, "nationkey", sqla.sql.sqltypes.BigInteger)
+    assert_column(nations, "name", sqla.sql.sqltypes.String)
+    assert_column(nations, "regionkey", sqla.sql.sqltypes.BigInteger)
+    assert_column(nations, "comment", sqla.sql.sqltypes.String)
+    query = sqla.select(nations)
+    result = conn.execute(query)
+    rows = result.fetchall()
+    assert len(rows) == 25
+    for row in rows:
+        assert isinstance(row['nationkey'], int)
+        assert isinstance(row['name'], str)
+        assert isinstance(row['regionkey'], int)
+        assert isinstance(row['comment'], str)
+
+
+def assert_column(table, column_name, column_type):
+    assert getattr(table.c, column_name).name == column_name
+    assert isinstance(getattr(table.c, column_name).type, column_type)
+
+
+@pytest.mark.parametrize('trino_connection', ['system'], indirect=True)
+def test_select_specific_columns(trino_connection):
+    _, conn = trino_connection
+    metadata = sqla.MetaData()
+    nodes = sqla.Table('nodes', metadata, schema='runtime', autoload_with=conn)
+    assert_column(nodes, "node_id", sqla.sql.sqltypes.String)
+    assert_column(nodes, "state", sqla.sql.sqltypes.String)
+    query = sqla.select(nodes.c.node_id, nodes.c.state)
+    result = conn.execute(query)
+    rows = result.fetchall()
+    assert len(rows) > 0
+    for row in rows:
+        assert isinstance(row['node_id'], str)
+        assert isinstance(row['state'], str)
+
+
+@pytest.mark.parametrize('trino_connection', ['memory'], indirect=True)
+def test_define_and_create_table(trino_connection):
+    engine, conn = trino_connection
+    if not engine.dialect.has_schema(engine, "test"):
+        engine.execute(sqla.schema.CreateSchema("test"))
+    metadata = sqla.MetaData()
+    try:
+        sqla.Table('users',
+                   metadata,
+                   sqla.Column('id', sqla.Integer),
+                   sqla.Column('name', sqla.String),
+                   sqla.Column('fullname', sqla.String),
+                   schema="test")
+        metadata.create_all(engine)
+        assert sqla.inspect(engine).has_table('users', schema="test")
+        users = sqla.Table('users', metadata, schema='test', autoload_with=conn)
+        assert_column(users, "id", sqla.sql.sqltypes.Integer)
+        assert_column(users, "name", sqla.sql.sqltypes.String)
+        assert_column(users, "fullname", sqla.sql.sqltypes.String)
+    finally:
+        metadata.drop_all(engine)
+
+
+@pytest.mark.parametrize('trino_connection', ['memory'], indirect=True)
+def test_insert(trino_connection):
```

Comment:
Can you please add tests for update and delete?

Reply:
Let's create some follow-up issues for that.